### PR TITLE
Login

### PR DESCRIPTION
**Pull request by AlfredWei0420**

### CODE REVIEW

File: src/client/board.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+/* class Home extends React.Component {
```
Comment (beltekylevi) [inline src/client/board.jsx]: do not leave comments

File: src/client/index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-import ReactDOM from 'react-dom';
-import './style.scss';
+import ReactDom from 'react-dom';
+import { BrowserRouter as Router, Route, Link, Switch, browserHistory } from 'react-router-dom'
+import App from './router.js'
 
-ReactDOM.render(
-  <h1>Hello, world!</h1>, document.querySelector('#root')
-);
+ReactDom.render(<App /> ,document.getElementById('root')
```
Comment (beltekylevi) [inline src/client/index.js]: Did you run the linter?!

File: src/client/login/footer.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+import React from 'react';
```
Comment (beltekylevi) [inline src/client/login/footer.js]: new line after `'use strict';`. run the linter! 😡

File: src/client/login/header.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+'use strict';
```
Comment (beltekylevi) [inline src/client/login/header.js]: new line after `'use strict';`. run the linter! 😡

File: src/client/login/header.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import './scss/header.scss';
+
+class Header extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    return (
+      <header>
+        <img src="../../img/black_pearl.png" alt=""/>
```
Comment (beltekylevi) [inline src/client/login/header.js]: Fill the alt attribute!

File: src/client/login/input.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+'use strict';
+import React from 'react';
+import './scss/input.scss';
+
+class Input extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+  // verify() {
```
Comment (beltekylevi) [inline src/client/login/input.js]: Don't leave comment!

File: src/client/login/input.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  //   } else if (password === '') {
+  //     alert('password cannot be none!');
+  //   } else if (password.length < 6) {
+  //     alert('password must be above 6 characters!');
+  //   }
+  // }
+  render() {
+    return (
+      <form className="input-form">
+        <input type="text" title="username" id="username" required placeholder="Username/email" ref="username" />
```
Comment (beltekylevi) [inline src/client/login/input.js]: What is the `title` attribute on `input`? 🤔

File: src/client/login/input.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        <input type="password" title="password" id="password" required placeholder="Password" ref="password" />
```
Comment (beltekylevi) [inline src/client/login/input.js]: What is the `ref` attribute on `input`? 🤔

File: src/client/login/login.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+'use strict';
```
Comment (beltekylevi) [inline src/client/login/login.js]: new line after 'use strict';. run the linter! 😡

File: src/client/login/scss/input.scss [hunk 10]
```diff
@@ -0,0 +1,33 @@
+.input-form {
+
```
Comment (beltekylevi) [inline src/client/login/scss/input.scss]: why do you have an empty line here? 🤔

File: src/client/login/scss/login.scss [hunk 11]
```diff
@@ -0,0 +1,3 @@
+#main{
```
Comment (beltekylevi) [inline src/client/login/scss/login.scss]: space between the selector name and the curly brackets

```css
#main {
...
}
```

File: src/client/router.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { BrowserRouter as Router, Route, Link, Switch, browserHistory } from 'react-router-dom'
```
Comment (beltekylevi) [inline src/client/router.js]: do you need `browserHistory`?

File: src/client/router.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+import Login from './login/login'
+import Board from './board.jsx'
+
+class App extends React.Component {
+  render() {
+    return (
+      <Router>
+        <div>
+          <Switch>
```
Comment (beltekylevi) [inline src/client/router.js]: can you explain to me what `switch` is good for?

File: src/server/index.js [hunk 14]
```diff
@@ -11,6 +11,10 @@ const app = express();
 
 app.use(express.static(path.resolve(__dirname, '../../dist')));
 
+app.get('/login', (req, res) => {
+  res.sendFile('index.html', { root: path.join(__dirname, '../../dist')});
```
Comment (beltekylevi) [inline src/server/index.js]: run the linter 😡

**Review by beltekylevi — CHANGES_REQUESTED** [review]

File: src/client/login/input.js (hunk 7 above)
Comment (AlfredWei0420) [reply src/client/login/input.js]: zeplin has a hover effect use that

File: src/client/login/input.js (hunk 8 above)
Comment (AlfredWei0420) [reply src/client/login/input.js]: future may grab the input value

File: src/client/Componets/login/index.js [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+class Login extends React.Component {
+  render() {
+    return (
+      <main>
+        <header>
+          <img src="../../img/black_pearl.png" alt="Black_pearl"/>
+          <h1>Welcome Onboard!</h1>
+        </header>
+        <LoginForm />
+        <p>Don`t have an account?<a href=""> Sign up>></a></p>
```
Comment (beltekylevi) [inline src/client/Componets/login/index.js]: `<p>Don`t have an account? <a href="#">Sign up>></a></p>`

also remove the `>>` from the end

File: src/client/Componets/login/index.scss [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+  font-family: MarkerFelt-Wide;
+}
+
+main {
+  text-align: center;
+  background-color: white;
+  padding: 20px 42px;
+  margin: 100px;
+  box-shadow: -1px 2px 126px -14px rgba(102,121,143,1);
+  p{
```
Comment (beltekylevi) [inline src/client/Componets/login/index.scss]: no empty css rules

File: src/client/Componets/login/index.scss [hunk 17]
```diff
@@ -0,0 +1,27 @@
+$headingColor: #67a2e7;
+
+h1 {
+  color: $headingColor;
+  font-family: MarkerFelt-Wide;
```
Comment (beltekylevi) [inline src/client/Componets/login/index.scss]: this will not work

File: src/client/Componets/loginForm/index.js [hunk 18]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import './index.scss';
+
+class LoginForm extends React.Component {
+  render() {
+    return (
+      <form className="login-form">
+        <input type="email" required placeholder="email" ref="username" />
```
Comment (beltekylevi) [inline src/client/Componets/loginForm/index.js]: remove refs

File: src/client/app/app.js [hunk 19]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import {BrowserRouter as Router, Route} from 'react-router-dom';
+import Login from '../Componets/login';
+
```
Comment (beltekylevi) [inline src/client/app/app.js]: empty line

File: src/client/Componets/login/index.js [hunk 20]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import './index.scss';
+import LoginForm from '../loginForm/index.js';
+
+class Login extends React.Component {
+  render() {
+    return (
+      <main>
+        <header>
+          <img src="../../img/black_pearl.png" alt="Black_pearl"/>
```
Comment (beltekylevi) [inline src/client/Componets/login/index.js]: example alt: The logo of Black Pearl

**Review by beltekylevi — CHANGES_REQUESTED** [review]
fix image

**Review by beltekylevi — APPROVED** [review]

**Pull request merged**